app/assets/javascripts/webauthn-registration.js

```diff
@@ -0,0 +1,106 @@
+var form = document.querySelector('.webauthn_key_registration')
```

Comment:
Needing to translate the GitHub TS to JS is a pain... and also a potential copyright worry.

From the lofty position of a backend developer who wouldn't do any of the work, I think the best solution would be for the GOV.UK frontend stuff to switch to webpack.  It would probably be a lot of work, but it's the way the JS ecosystem has been moving for a while now, and TS is only getting more common.

On the other hand, that's not going to happen for a while.  So maybe we do just have to write our own WebAuthn client-side implementation.

This could even be a point against using Rails / govuk_publishing_components.  There are service teams using the Design System outside of Ruby.

Reply:
ah super interestng, Hadn't considered the copyright angle 😬.
Yeah i 💯 agree!